python/src/test/java/org/apache/zeppelin/python/PythonInterpreterTest.java

```diff
@@ -164,13 +166,16 @@ public void testInterpret() {
    * @param port
    */
   private boolean serverIsListeningOn(Integer port) {
-    boolean serverIsListening = false;
     Socket s = new Socket();
+    boolean serverIsListening = false;
 
-    boolean connected = tryToConnect(s, port);
-    if (connected) {
-      serverIsListening = true;
+    int retryCount = 0;
+    boolean connected = false;
+    while (connected = tryToConnect(s, port) && retryCount < 10) {
+      serverIsListening = connected;
       tryToClose(s);
+      retryCount++;
+      s = new Socket();
```

Comment:
`sleep` added inside the client code in tests, where we know that it's needed to wait for something to startup\shutdown